api/proto/attest/attest.proto

```diff
@@ -0,0 +1,58 @@
+// Copyright(c) 2017-2018 Zededa, Inc.
+// All rights reserved.
+
+syntax = "proto3";
+
+option go_package  = "github.com/lf-edge/eve/api/go/certs";
+option java_package = "com.zededa.cloud.uservice.proto";
+
+// This is the response payload for POST /api/v1/edgeDevice/attestNonce
+// or /api/v2/edgeDevice/attestNonce
+// The message is assumed to be protected by a TLS session bound to the
+// device certificate for v1.
+// The message is assumed to be protected by signing envelope for v2.
+
+message ZAttestNonceMsg {
+  ZAttestDataTypes type = 1;  //sub-type, mostly for future proofing
+  ZNonce nonce = 2;           //nonce message content
+}
+
+message ZNonce {
+  string nonce = 1;  //nonce to use in quote generation
+}
+
+message ZAttestQuoteMsg {
+  ZAttestDataTypes type = 1; //sub-type, mostly for future-proofing
+  ZAttestQuote quote = 2; // the actual quote from TPM
+}
+
+// This is the request payload for POST /api/v1/edgeDevice/attestQuote
+// or /api/v2/edgeDevice/attestQuote
+// The message is assumed to be protected by a TLS session bound to the
+// device certificate for v1.
+// The message is assumed to be protected by signing envelope for v2.
+message ZAttestQuote {
+  string devUuId = 1;  //device UUID
+  string attestData = 2; //TPMS_ATTEST (Table 2:123) in https://trustedcomputinggroup.org/wp-content/uploads/TPM-Rev-2.0-Part-2-Structures-01.38.pdf
```

Comment:
In the Nonce inside the attestData?

Reply:
Yes, it is part of attestData. I've added a comment there to reflect this.